### PR TITLE
Clean up TransactionReport and AccountTransactionReport.

### PR DESCRIPTION
**Pull request by Xitian9**
`transactionsReport` was only used in one place, and a close look revealed it was exactly equivalent to `entriesReport`. This allows us to remove the `TransactionsReport` module entirely. In addition to being cleaner, it also removes a distinction that always confused me (when should I use AccountTransactionsReport, TransactionsReport, and EntriesReport?).

We then generate the overall report query for account transactions reports within `accountTransactionsReport`, rather than being generated externally and then supplied as an argument. This brings it in line with the practice in other reports, reduces duplication, and increases clarity and uniformity.

### CODE REVIEW

**Comment by Xitian9** [comment]
Since this is a blocker for moving forward on the lenses, let me know if there's anything I can do to make this easier to review.

**Comment by simonmichael** [comment]
Will do. It's next on my list.

**Review by simonmichael — COMMENTED** [review]
Doh. I hadn't submitted this question.

File: hledger-web/templates/journal.hamlet [hunk 1]
```diff
@@ -14,15 +14,13 @@ $if elem CapAdd caps
...
-    $forall (torig, _, split, _, amt, _) <- items
+    $forall torig <- items
       <tr .title ##{transactionFrag torig} title="#{showTransaction torig}">
         <td .date nowrap>
           #{show (tdate torig)}
         <td colspan=2>
           #{textElideRight 60 (tdescription torig)}
         <td .amount style="text-align:right;">
-          $if not split && not (mixedAmountLooksZero amt)
-            ^{mixedAmountAsHtml amt}
```
[the remark this answers is not in the document]
Comment (Xitian9) [reply hledger-web/templates/journal.hamlet:25]: I did. The second dot point of the first commit of this PR explains why `mixedAmountLooksZero amt` is always `True`, and so this code will never fire.

[the remark this answers is not in the document]
Comment (simonmichael) [reply hledger-web/templates/journal.hamlet:25]: I was scratching my head.. thanks!

**Pull request merged**